src/Arduino_EthernetConnectionHandler.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+   This file is part of ArduinoIoTCloud.
+   Copyright 2020 ARDUINO SA (http://www.arduino.cc/)
+   This software is released under the GNU General Public License version 3,
+   which covers the main part of arduino-cli.
+   The terms of this license can be found at:
+   https://www.gnu.org/licenses/gpl-3.0.en.html
+   You can be released from the requirements of the above licenses by purchasing
+   a commercial license. Buying such a license is mandatory if you want to modify or
+   otherwise use the software for commercial activities involving the Arduino
+   software without disclosing the source code of your own applications. To purchase
+   a commercial license, send an email to license@arduino.cc.
+*/
+
+/******************************************************************************
+   INCLUDE
+ ******************************************************************************/
+
+#include "Arduino_EthernetConnectionHandler.h"
+
+#ifdef BOARD_HAS_ETHERNET /* Only compile if the board has ethernet */
+
+/******************************************************************************
+   CTOR/DTOR
+ ******************************************************************************/
+
+EthernetConnectionHandler::EthernetConnectionHandler(bool const keep_alive)
+: ConnectionHandler{keep_alive, NetworkAdapter::ETHERNET}
+,_ip{INADDR_NONE}
+,_dns{INADDR_NONE}
+,_gateway{INADDR_NONE}
+,_subnet{INADDR_NONE}
+{
+
+}
+
+EthernetConnectionHandler::EthernetConnectionHandler(IPAddress ip, IPAddress dns, IPAddress gateway, IPAddress subnet, bool const keep_alive)
+: ConnectionHandler{keep_alive, NetworkAdapter::ETHERNET}
+,_ip{ip}
+,_dns{dns}
+,_gateway{gateway}
+,_subnet{subnet}
+{
+
+}
+
+/******************************************************************************
+   PROTECTED MEMBER FUNCTIONS
+ ******************************************************************************/
+
+NetworkConnectionState EthernetConnectionHandler::update_handleInit()
+{
+  if (Ethernet.hardwareStatus() == EthernetNoHardware) {
+#if !defined(__AVR__)
```

Comment:
This comment is not explicitly targeted on this PR. But since AVR is not and will not be supported on Arduino IoT Cloud (its not technically feasible), you can drop all those `#if !defined(__AVR__)`statements.